### PR TITLE
Adds a license press for in perma brig

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
add: Adds a license press, make those prisoners do something useful!
/:cl:

adds a license press prisoners can use to make license plates from empty ones. Makes a small profit

vid: https://streamable.com/z64rl

### CODE REVIEW

**Comment by Bawhoppen** [comment]
thank you dearest floyd

File: code/game/objects/items/stacks/license_plates.dm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	max_amount = 20
+
+/obj/item/stack/license_plates/empty
+	name = "empty license plate"
+	desc = "Instead of a license plate number, this could contain a quote like \"Live laugh love\""
+
+/obj/item/stack/license_plates/filled
+	name = "license plate"
+	desc = "Prison labor paying off"
+	icon_state = "filled_plate_1_1"
```
Comment (ATH1909) [inline code/game/objects/items/stacks/license_plates.dm]: P E R I O D S

**Comment by ATH1909** [comment]
So, uh, how exactly are you supposed to gain money from this? Can Cargo sell these plates to CentCom?

**Comment by Bawhoppen** [comment]
also the sprites are from me and JustRandomGuy

@ATH1909 they will be exportable and that can easily be added in an accompanying PR if not in this one

**Comment by Qustinnus** [comment]
i was adding it already but had already made the pr

**Comment by ATH1909** [comment]
Will these be in the permabrig at roundstart?

**Comment by Farquaar** [comment]
They should probably be too big to fit in people's pockets if they aren't already.

**Comment by Qustinnus** [comment]
how do you out a license press in your pocket

**Comment by Farquaar** [comment]
> 
> 
> how do you out a license press in your pocket

You know what I mean you smelly lightbulb enthusiast

**Comment by Cyberboss** [comment]
> Prison Architect flashbacks

File: code/game/machinery/prisonlabor.dm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	icon_state = "offline"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 2
+	active_power_usage = 50
+	var/obj/item/stack/license_plates/empty/current_plate
+	var/pressing = FALSE
+
+/obj/machinery/plate_press/update_icon()
+	. = ..()
+	if(machine_stat & (BROKEN|NOPOWER))
```
Comment (Cyberboss) [inline code/game/machinery/prisonlabor.dm]:
```suggestion
	if(!is_operational())
```

File: code/game/machinery/prisonlabor.dm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		icon_state = "offline"
+	else if(pressing)
+		icon_state = "loop"
+	else if(current_plate)
+		icon_state = "online_loaded"
+	else
+		icon_state = "online"
+
+/obj/machinery/plate_press/attackby(obj/item/I, mob/living/user, params)
+	if(machine_stat & (BROKEN|NOPOWER))
```
Comment (Cyberboss) [inline code/game/machinery/prisonlabor.dm]:
```suggestion
	if(!is_operational())
```

File: code/game/machinery/prisonlabor.dm [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/obj/machinery/plate_press
+	name = "license plate press"
+	desc = "You know, we're making a lot of license plates for a station with literaly no cars in it."
+	icon = 'icons/obj/machines/prison.dmi'
+	icon_state = "offline"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 2
+	active_power_usage = 50
+	var/obj/item/stack/license_plates/empty/current_plate
```
Comment (Cyberboss) [inline code/game/machinery/prisonlabor.dm]: Needs cleanup in `/Destroy()`

**Review by Cyberboss — CHANGES_REQUESTED** [review]
These files look like they need a linter run.

File: code/game/machinery/prisonlabor.dm [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		update_icon()
+	else
+		return ..()
+
+/obj/machinery/plate_press/attack_hand(mob/living/user)
+	. = ..()
+	if(!pressing && current_plate)
+		work_press(user)
+
+/obj/machinery/plate_press/proc/work_press(mob/living/user)
```
Comment (Cyberboss) [inline code/game/machinery/prisonlabor.dm]: autodoc

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Add autodocs too

**Comment by OverlyGeneric** [comment]
Ah yes, bureaucracy for the bureaucrat gods.

**Comment by cacogen** [comment]
Maybe I'm just unimaginative but this seems out of place and Goon-like.

**Review by spookydonut — APPROVED** [review]

**Pull request merged**